packages/node/test/handlers.test.ts

```diff
@@ -498,4 +527,28 @@ describe('errorHandler()', () => {
     const requestSession = scope?.getRequestSession();
     expect(requestSession).toEqual(undefined);
   });
+
+  it('stores request in `sdkProcessingMetadata`', () => {
+    const options = getDefaultNodeClientOptions({});
+    client = new NodeClient(options);
+
+    const hub = new Hub(client);
+    makeMain(hub);
+
+    // `sentryErrorMiddleware` uses `withScope`, and we need access to the temporary scope it creates, so we monkeypatch
+    // `pushScope` in order to store the new scope it creates. (This initial value for `scope` will be replaced when `pushScope` is called by
+    // `withScope` inside of `sentryErrorMiddleware`, and is just here to prevent the `expect` statement below from
+    // throwing a `Variable 'scope' is used before being assigned` error.)
+    let scope: Scope = {} as Scope;
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    const origPushScope = hub.pushScope;
+    hub.pushScope = function (this: Hub) {
+      scope = origPushScope.call(this);
+      return scope;
```

Comment:
m: Why can't we just monkeypatch `withScope`? This will break if `withScope` stops uses `pushScope` in the future, which it might.

Reply:
That (wrapping `withScope` instead) was my first thought, actually, but in order to do so I would have to copy its code rather than wrap it, because the scope in question isn't returned:

```js
  // the real version
  public withScope(callback: (scope: Scope) => void): void {
    const scope = this.pushScope();
    try {
      callback(scope);
    } finally {
      this.popScope();
    }
  }
```

So I'd have to do

```js
  // monkeypatched version
  hub.withScope = function(this:Hub, callback: (scope: Scope) => void): void {
    scope = this.pushScope();
    try {
      callback(scope);
    } finally {
      this.popScope();
    }
  }
```

which, to be fair, isn't _that_ much more code, but a) is at least a little uglier than the `pushScope` monkeypatching, and more importantly, b) doesn't save us from having to fix the test if the implementation of `withScope` changes. Given that, monkeypatching `pushScope` seemed a better option.

Reply:
Why do we need to replicate functionality? We just need to grab a reference to the scope right (which should get mutated).

Reply:
Then can we try something like so?

```ts
// eslint-disable-next-line @typescript-eslint/unbound-method
const oldCaptureException = hub.captureException;
hub.captureException = function (this: Hub, ...args: unknown[]) {
  const scope = this.getScope();
  expect((scope as any)._sdkProcessingMetadata.request).toEqual(req);
  return oldCaptureException.call(this, ...args);
};
```

Which, basically checks if the scope was correct at the time of `captureException`.